text/2978-stack_based_vec.md

```diff
@@ -0,0 +1,364 @@
+
+    pub const fn capacity(&self) -> usize;
+
+    pub fn clear(&mut self);
+
+    pub const fn is_empty(&self) -> bool;
+
+    pub const fn len(&self) -> usize;
+
+    pub fn retain<F>(&mut self, mut f: F)
+    where
+        F: FnMut(&mut T) -> bool;
+
+    pub fn truncate(&mut self, len: usize);
+
+    // Methods that can panic at run-time
+
+    pub fn drain<R>(&mut self, range: R) -> Drain<'_, T, N>
+    where
+        R: RangeBounds<usize>;
+
+    pub fn extend_from_cloneable_slice<'a>(&mut self, other: &'a [T])
+    where
+        T: Clone;
+
+    pub fn extend_from_copyable_slice<'a>(&mut self, other: &'a [T])
+    where
+        T: Copy;
+
+    pub const fn insert(&mut self, idx: usize, element: T);
+
+    pub const fn push(&mut self, element: T);
+
+    pub const fn remove(&mut self, idx: usize) -> T;
+
+    pub fn splice<I, R>(&mut self, range: R, replace_with: I) -> Splice<'_, I::IntoIter, N>
+    where
+        I: IntoIterator<Item = T>,
+        R: RangeBounds<usize>;
+
+    pub fn split_off(&mut self, at: usize) -> Self;
+
+    pub fn swap_remove(&mut self, idx: usize) -> T;
+
+    // Verifiable methods
+    
+    pub const fn pop(&mut self) -> Option<T>;
+}
+```
+
+Meaningless, unstable and deprecated methods like `reserve` or `drain_filter` weren't considered. A concrete implementation is available at https://github.com/c410-f3r/stack-based-vec.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+### Additional complexity
+
+New and existing users are likely to find it difficult to differentiate the purpose of each vector type, especially people that don't have a theoretical background in memory management.
+
+### The current ecosystem is fine
+
+`ArrayVec` might be an overkill in certain situations. If someone wants to use stack memory in a specific application, then it is just a matter of grabbing the appropriated crate.
+
+# Prior art
+[prior-art]: #prior-art
+
+These are the most known structures:
+
+ * `arrayvec::ArrayVec`: Uses declarative macros and an `Array` trait for implementations but lacks support for arbitrary sizes.
+ * `heapless::Vec`: With the usage of `typenum`, can support arbitrary sizes without a nightly compiler.
+ * `staticvec::StaticVec`: Uses unstable constant generics for arrays of arbitrary sizes.
+ * `tinyvec::ArrayVec`: Supports fixed and arbitrary (unstable feature) sizes but requires `T: Default` for security reasons.
+
+As seen, there isn't an implementation that stands out among the others because all of them roughly share the same purpose and functionality. Noteworthy is the usage of constant generics that makes it possible to create an efficient and unified approach for arbitrary array sizes.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+### Verifiable methods
+
+Unlike methods that will abort the current thread execution, verifiable methods will signal that something has gone wrong or is missing. This approach has two major benefits:
```

Comment:
This should also list downsides, e.g. having an interface inconsistent with `Vec`.